### PR TITLE
MNTOR-1395: add async script for recencyToBreachId

### PR DESCRIPTION
**Pull request by mansaj**
# References: 
Jira: MNTOR-1395
Figma: 

# Description
Switch recencyToBreachId to async mode so it can be ran as a kube job

# Screenshot (if applicable)

Not applicable.

# How to test

# Checklist (Definition of Done)
- [ ] Localization strings (if needed) have been added.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/scripts/kube-jobs/recencyToBreachId.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        .update({
+          db_migration_2: true
+        })
+        .transacting(trx)
+    }))
+
+    await trx.commit()
+  } catch (error) {
+    await trx.rollback()
+    console.log('select & mark rows failed!! first row:')
```
Comment (pdehaan) [inline src/scripts/kube-jobs/recencyToBreachId.js:49]: Should this be `console.log()` or `console.error()`?
Minor nit, but noticed that you used `console.error()` below on L77:

https://github.com/mozilla/blurts-server/blob/28c393374ca69d9733d8605c0f7a5d56669287a1/src/scripts/kube-jobs/recencyToBreachId.js#L77

File: src/scripts/kube-jobs/recencyToBreachId.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+// find all subscribers who resolved any breaches in the past,
+// replace recency index with breach id
+
+let failedToSelect = true
+while (failedToSelect) {
+  try {
+    subscribersArr = await selectAndLockResolutions()
+    failedToSelect = false
+  } catch (e) {
+    failedToSelect = true
```
Comment (pdehaan) [inline src/scripts/kube-jobs/recencyToBreachId.js:101]: I think this is already set to `true` above on L95.
But does this need some sort of sleep or backoff or total number of attempts before launching itself into the sun?
Not sure how possible it is that `selectAndLockResolutions()` might fail and put this in an infinite loop.

File: src/scripts/kube-jobs/recencyToBreachId.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+import { getAllBreachesFromDb } from '../../utils/hibp.js'
+import { getAllEmailsAndBreaches } from '../../utils/breaches.js'
+const knex = Knex(knexConfig)
+
+const LIMIT = 1000 // with millions of records, we have to load a few at a time
+let subscribersArr = []
+
+const selectAndLockResolutions = async () => {
+  const trx = await knex.transaction()
+  let subscribersArr = []
```
Comment (pdehaan) [inline src/scripts/kube-jobs/recencyToBreachId.js:24]: nit: not sure if you want to rename this w/ an underscore or something to disambiguate it from L20 above w/ the same name. Took me a second to figure out where the L20 variable was getting set from. Spoiler alert: L98

https://github.com/mozilla/blurts-server/blob/28c393374ca69d9733d8605c0f7a5d56669287a1/src/scripts/kube-jobs/recencyToBreachId.js#L98

Comment (mansaj) [reply src/scripts/kube-jobs/recencyToBreachId.js:24]: oh good catch, there should just be one

File: src/scripts/kube-jobs/recencyToBreachId.js (hunk 1 above)
Comment (mansaj) [reply src/scripts/kube-jobs/recencyToBreachId.js]: Fixed, thanks

File: src/scripts/kube-jobs/recencyToBreachId.js (hunk 2 above)
Comment (mansaj) [reply src/scripts/kube-jobs/recencyToBreachId.js]: Fixed, thanks

Comment (mansaj) [reply src/scripts/kube-jobs/recencyToBreachId.js]: The select is expected to return without error even if we run out of records. The only way it should error out is if the update (lock marking) failed due to race condition. We haven't seen this happen during our tests due to these kubernetes pods being spun up with a slight delay

File: src/scripts/kube-jobs/recencyToBreachId.js (hunk 3 above)
Comment (mansaj) [reply src/scripts/kube-jobs/recencyToBreachId.js]: Fixed. Thanks!

**Pull request merged**